File: craps.py
from random import *

def gera_dado ():
  return randint(1, 6), randint(1, 6)

def inicio():
  return print("\n\n=============== CRAPS ================")

def estagio_um():
    dado1, dado2 = gera_dado()
    pontos = dado1 + dado2
  
    if (pontos == 7) or (pontos == 11):
       print("\nDado 1: ", dado1, "\nDado 2: ", dado2, "\nPonto: ", pontos, "\n\nVocê ganhou!!!\n\n")  
       sentinela = int(input("\n Voce quer continuar? (0 para sair e 1 para continuar): "))
    elif (pontos == 2) or (pontos == 3) or (pontos == 12):
       print("\nDado 1: ", dado1, "Dado 2: ", dado2, "\nPonto: ", pontos, "\n\nVoce perdeu!")
       sentinela = int(input("\n Voce quer continuar? (0 para sair e 1 para continuar): "))
   
    else:
       print("\nDado 1: ", dado1, "Dado 2: ", dado2, "\nPonto: ", pontos)
       print("\n\nIniciando estagio 2...")
       secundo_estagio(pontos)

def secundo_estagio(meta):
     jogada = 1
     dado1, dado2 = gera_dado()
     pontos = dado1 + dado2

     if (pontos == 7):
       print("\nJogada: ", jogada, "\nDado 1: ", dado1, "Dado 2: ", dado2, "\nPonto: ", pontos)
       print("\n\nVoce perdeu!!")
       sentinela = int(input("\nVoce quer continuar? (0 para sair e 1 para continuar): "))

     elif (pontos == meta):
       print("\n\nJogada: ", jogada, "\nDado 1: ", dado1, "Dado 2: ", dado2, "\nPontos: ", pontos, "\n\nVoce ganhou!!")
       sentinela = int(input("\nVoce quer continuar? (0 para sair e 1 para continuar): "))

     else: 
       print("\n\nJogada: ", jogada, "\nDado 1: ", dado1, "Dado 2: ", dado2, "\nPontos: ", pontos)
      
     jogada+1
     
sentinela = 1

while (sentinela == 1): 
  if (sentinela == 1):
    inicio()
    estagio_um()
  else:
    exit(-1) 

File: craps.java
import java.util.Random;

public class Main {
  public static void main(String[] args) {
     Random gerador = new Random ();

    int jogadas[] = new int[2];//guardar os valores dos dois dados
    int dado, jogador, ponto;//jogador é a soma dos dois dados

     System.out.println("\n=============== CRAPS ================");
     dado = gerador.nextInt(6) + 1;
     jogadas[0]  = dado;
     dado = gerador.nextInt(6) + 1;
     jogadas[1] = dado;
     jogador = jogadas[0] + jogadas[1];

    if (jogador == 7 || jogador == 11){
      System.out.printf("\nDado 1: %d, Dado 2: %d.\nPonto: %d", jogadas[0], jogadas[1], jogador);
   
      System.out.println("\nVoce ganhou!"); 
    } else if (jogador == 2 || jogador == 3 || jogador == 12){
      System.out.printf("\nDado 1: %d, Dado 2: %d.\nPonto: %d", jogadas[0], jogadas[1], jogador);
      System.out.println("\nVoce perdeu!");
   }
    else {
      
       System.out.printf("\nDado 1: %d, Dado 2: %d.\nPonto: %d", jogadas[0], jogadas[1], jogador);
      ponto = jogador;
       System.out.println("\n\nIniciando estagio 2...");
      
      for (int i = 1;;i++){
        
        dado = gerador.nextInt(6) + 1;
        jogadas[0]  = dado;
        dado = gerador.nextInt(6) + 1;
        jogadas[1] = dado;
        jogador = jogadas[0] + jogadas[1];

         System.out.printf("\nJogada: %d\nDado 1: %d, Dado 2: %d.\nPonto: %d\n", i, jogadas[0], jogadas[1], jogador);
        
        if(jogador == 7){
         System.out.println("Voce perdeu!");
          return;
        }
        
        if (jogador == ponto){
         System.out.println("Voce ganhou!");
          return;
        }
      }
    }  
  }
}
